packages/shared/src/components/utilities/DateFormat.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { ReactElement, useMemo } from 'react';
+import { formatDate, TimeFormatType } from '../../lib/dateFormat';
+import { useFeature } from '../GrowthBookProvider';
+import { feature } from '../../lib/featureManagement';
+import { PublishTimeFormat } from '../../lib/featureValues';
+
+interface DateFormatProps {
+  date: string | number | Date;
+  type?: TimeFormatType;
+  className?: string;
+  prefix?: string;
+}
+export default function DateFormat({
+  date,
+  type,
+  className,
+  prefix,
+}: DateFormatProps): ReactElement {
+  const publishTimeFormat = useFeature(feature.publishTimeFormat);
+  const timeFormat =
+    publishTimeFormat === PublishTimeFormat.V1 &&
+    type !== TimeFormatType.ReadHistory
+      ? TimeFormatType.Generic
+      : type;
+
+  const renderDate = useMemo(
+    () => date && formatDate({ value: date, type: timeFormat }),
+    [date, timeFormat],
+  );
+
+  return (
+    <time className={className} dateTime={date.toString()}>
```

Comment:
Should we add a `title` to this too? So you can hover over it to see the actual date?
![image](https://github.com/dailydotdev/apps/assets/1681525/1b1d860c-55fc-46fc-ab59-2124022d4ef1)

Also, I think `dateTime` should be ISO string? `toString()` isn't really a valid `datetime` string iirc

Reply:
Let me check the specs :)